dataland-frontend/src/components/generics/ViewFrameworkBase.vue

```diff
@@ -178,8 +178,9 @@ export default defineComponent({
           this.singleDataMetaInfoToDisplay.dataId
         );
       } else if (this.mapOfReportingPeriodToActiveDataset.size > 1 && !this.singleDataMetaInfoToDisplay) {
-        if (this.$refs.reportingPeriodsOverlayPanel instanceof OverlayPanel) {
-          this.$refs.reportingPeriodsOverlayPanel.toggle(event);
+        const panel = this.$refs.reportingPeriodsOverlayPanel as OverlayPanel;
+        if (panel) {
+          panel.toggle(event);
         }
       } else if (this.mapOfReportingPeriodToActiveDataset.size == 1 && !this.singleDataMetaInfoToDisplay) {
         this.gotoUpdateForm(
```

Comment:
Noticed that in line 189 we are overcomplicating things a little bit.

To retrieve the dataId there, we could simply write
`Array.from(mapOfReportingPeriodToActiveDataset.values())[0].dataId`

I find this easier to read.